### DISCUSSION TITLE
Join VDB files tool

### DISCUSSION THREAD

**Discussion by chrisflesher** [AcademySoftwareFoundation/openvdb#1586] [General]
I was interested in manipulating multiple VDB files using AX but the program only accepts one input file.

So far I've been joining VDB files together using a python script before applying AX. This seems to work fine for now, though I'm a bit concerned about running out of memory in the future.

Was wondering if you guys would consider providing a command line tool to join VDB files before applying AX (for convenience)? Am not sure how you would resolve conflicts due to multiple grids having the same name though...

**Comment by Idclip** [comment]
Hey Chris! So I think bits of the behaviour you're after (multiple files and name bindings) are available in `10.0.0`.

In `9.1.0` we added:

    - Added a --thread [n] argument to the vdb_ax binary.
    - Added a --copy-file-metadata option to vdb_ax. This behaviour is now
      off by default.
    - Added support for multiple input files with the vdb_ax binary using -i.
      Positional arguments as input files are deprecated.

And in `10.0.0` I re-worked the CLI argument parsing framework which exposed the majority of previously internal executable options like time streaming, iterator types and, importantly, name bindings. So with the latest release you can do something like:

`vdb_ax -i cube.vdb -i sphere.vdb -s "@a=max(@a,@b);" -o out.vdb --bindings "a:ls_cube,b:ls_sphere" --opt O3 --node-iter ALL`

Unfortunately the name bindings are currently global; if you had the same named grid "ls_cube" in two different files, the binding would apply to both. We'll probably add something to handle this, maybe simply adding another options to `--bindings` e.g. `"filename:ax_name:vdb_name"`

**Comment by chrisflesher** [comment]
Perfect, thank you!!
